gcsfuse init fails with "panic: Page size is unexpectedly (big number)" with macFUSE + arm64/m1

* macOS 11.5.1 (20G80)
* arm64 / apple m1 silicon
* macFUSE version 4.1.2, installed with `pkg`
* built `go` from source
* built `gcsfuse` from source w/ GOPATH=$HOME/go
* and:
```
➜  ~ which go
/usr/local/go/bin/go
➜  ~ go version
go version go1.16.6 darwin/arm64
➜  ~ which gcsfuse
/Users/william/go/bin/gcsfuse
➜  ~ gcsfuse
panic: Page size is unexpectedly 16384

goroutine 1 [running]:
github.com/googlecloudplatform/gcsfuse/vendor/github.com/jacobsa/fuse/internal/buffer.init.0()
        /Users/william/go/src/github.com/googlecloudplatform/gcsfuse/vendor/github.com/jacobsa/fuse/internal/buffer/in_message.go:33 +0xb8
```

any idea what could be causing this?

there's a related issue with macFUSE 4.1.0 (https://github.com/GoogleCloudPlatform/gcsfuse/issues/514) but that seems to be an issue with naming or looking up the name of the macFUSE dependency. my problem seems more system related. 

maybe yet another arm64 compat issue?

--- a/go.mod
+++ b/go.mod
@@ -6,7 +6,7 @@ require (
 	cloud.google.com/go/storage v1.16.0
 	github.com/go-kit/kit v0.10.0 // indirect
 	github.com/jacobsa/daemonize v0.0.0-20160101105449-e460293e890f
-	github.com/jacobsa/fuse v0.0.0-20210805062233-5d146b000c70
+	github.com/jacobsa/fuse v0.0.0-20210818065549-10d864429bf7
 	github.com/jacobsa/gcloud v0.0.0-20210611094250-93436725bc18
 	github.com/jacobsa/oglematchers v0.0.0-20150720000706-141901ea67cd
 	github.com/jacobsa/oglemock v0.0.0-20150831005832-e94d794d06ff

